openrtb_ext/request_wrapper.go

```diff
@@ -791,6 +797,19 @@ func (re *RegExt) marshal() (json.RawMessage, error) {
 		re.usPrivacyDirty = false
 	}
 
+	if re.GDPRDirty {
+		gdprJson, err := json.Marshal(*re.GDPR)
```

Comment:
This can result in a panic. You should only call Marshal if `re.gdpr` is not nil, otherwise delete it from the map. This is a pattern we follow in other cases so you look around this file for an example.

Reply:
Corrected